apps/cowswap-frontend/src/modules/ordersTable/pure/OrdersTableContainer/TableGroup.tsx

```diff
@@ -28,6 +30,78 @@ const Pagination = styled(OrdersTablePagination)`
   padding: 10px 0;
 `
 
+const TwapStatusAndToggleWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 8px;
+  width: 100%;
+`
+
+function TwapStatusAndToggle({
+  parent,
+  childrenLength,
+  isCollapsed,
+  onToggle,
+  onClick,
+  children,
+}: {
+  parent: any
+  childrenLength: number
+  isCollapsed: boolean
+  onToggle: () => void
+  onClick: () => void
+  children: any[]
+}) {
+  // Check if any child has insufficient balance or allowance
+  const hasChildWithWarning = children.some(
+    (child) =>
+      (child.orderParams?.hasEnoughBalance === false || child.orderParams?.hasEnoughAllowance === false) &&
+      (child.order.status === OrderStatus.PENDING || child.order.status === OrderStatus.SCHEDULED),
+  )
+
+  // Get the first child with a warning to use its parameters
+  const childWithWarning = hasChildWithWarning
+    ? children.find(
+        (child) =>
+          (child.orderParams?.hasEnoughBalance === false || child.orderParams?.hasEnoughAllowance === false) &&
+          (child.order.status === OrderStatus.PENDING || child.order.status === OrderStatus.SCHEDULED),
+      )
+    : null
```

Comment:
isn't this the same as `hasChildWithWarning`?  so `hasChildWithWarning = !!childWithWarning`